assets/app/scripts/services/auth.js

```diff
@@ -276,4 +276,18 @@ angular.module('openshiftConsole')
       }
     }
   };
-}]);
+}])
+
+/* A WebSocket factory for kubernetesContainerTerminal */
+.factory("AuthContainerWebSocket", function(API_CFG, AuthService) {
```

Comment:
Because the model does not fit with what container-terminal needs. It needs a factory of WebSocket() like objects. $ws returns promises. I'd rather not export that complexity to other projects using the container-terminal widget.